src/Symfony/Component/String/AbstractString.php

```diff
@@ -0,0 +1,555 @@
+            throw new InvalidArgumentException('Invalid UTF-8 chars.');
+        }
+        $chars = preg_quote($chars);
+
+        $str = clone $this;
+        $str->string = preg_replace("{^[$chars]++}u", '', $str->string);
+
+        return $str;
+    }
+
+    /**
+     * @return static
+     */
+    public function trimRight(string $chars = " \t\n\r\0\x0B\x0C\u{A0}\u{FEFF}"): self
+    {
+        if (!preg_match('//u', $chars)) {
+            throw new InvalidArgumentException('Invalid UTF-8 chars.');
+        }
+        $chars = preg_quote($chars);
+
+        $str = clone $this;
+        $str->string = preg_replace("{[$chars]++$}u", '', $str->string);
+
+        return $str;
+    }
+
+    /**
+     * @return static
+     */
+    public function truncate(int $length, string $ellipsis = ''): self
+    {
+        $stringLength = $this->length();
+
+        if ($stringLength <= $length) {
+            return $this;
+        }
+
+        $ellipsisLength = '' !== $ellipsis ? (new static($ellipsis, $this->locale))->length() : 0;
+
+        if ($length < $ellipsisLength) {
+            $ellipsisLength = 0;
+        }
+
+        $str = $this->substr(0, $length - $ellipsisLength);
+
+        return $ellipsisLength ? $str->trimRight()->append($ellipsis) : $str;
+    }
+
+    /**
+     * @return static
+     */
+    public function wordwrap(int $width = 75, string $break = "\n", bool $cut = false): self
+    {
+        $lines = $this->split($break);
+        $chars = [];
+        $mask = '';
+
+        if (1 === \count($lines) && '' === $lines[0]->string) {
+            return $lines[0];
+        }
+
+        foreach ($lines as $i => $line) {
+            if ($i) {
+                $chars[] = $break;
+                $mask .= '#';
+            }
+
+            foreach ($line->chunk() as $char) {
+                $chars[] = $char->string;
+                $mask .= ' ' === $char->string ? ' ' : '?';
+            }
+        }
+
+        $string = '';
+        $j = 0;
+        $b = $i = -1;
+        $mask = wordwrap($mask, $width, '#', $cut);
+
+        while (false !== $b = strpos($mask, '#', $b + 1)) {
+            for (++$i; $i < $b; ++$i) {
+                $string .= $chars[$j];
+                unset($chars[$j++]);
+            }
+
+            if ($break === $chars[$j] || ' ' === $chars[$j]) {
+                unset($chars[$j++]);
+            }
+
+            $string .= $break;
+        }
+
+        $str = clone $this;
+        $str->string = $string.implode('', $chars);
+
+        return $str;
+    }
+
+    /**
+     * @return static
+     */
+    public function withLocale(string $locale): self
+    {
+        $str = clone $this;
+        $str->locale = str_replace('-', '_', strtolower($locale));
+
+        return $str;
+    }
+
+    public function jsonSerialize(): string
+    {
+        return $this->string;
+    }
+
+    public function __clone()
+    {
+        $this->ignoreCase = false;
+    }
+
+    public function __toString(): string
```

Comment:
maybe we should add a `toString()` alias here ? `$obj->equals($obj2->toString())` is more friendly than `$obj->equals((string)$obj2)` or `$obj->equals($obj2->jsonSerialize())`

this component should also probably include a `Stringable` interface too : 

```php
interface Stringable
{
  public function toString(): string;
}
```

Reply:
This component provides `AbstractString`, no need for `Stringable` (which is a core php concern and where it should be added to me, for `__toString` btw)

I'm not super convinced here. You could use `__toString()` directly also.